### DISCUSSION TITLE
ejabberd as 'microservice' behind haproxy

### DISCUSSION THREAD

**Discussion by f1-outsourcing** [processone/ejabberd#4293] [Questions]
Is there some best practice, howto manual how to setup ejabberd in an environment where it is behind proxies? Currently I am running prosody just fine for incoming client connections. Problem is with the proxy traffic for outgoing s2s. I have a different socks proxy for outgoing traffic.

https://github.com/sando38/helm-ejabberd

**Comment by f1-outsourcing** [comment]
Everyone is running this directly connected to the public internet?

**Reply by licaon-kter** [reply]
https://www.process-one.net/blog/proxy-protocol-in-ejabberd/ helpful?

**Reply by f1-outsourcing** [reply]
I think this looks similar to my current prosody config. That is fine for incoming client connections and clients writing on the same server, however I have some issues with outgoing to other domains.

**Reply by f1-outsourcing** [reply]
I can't really believe that nobody is running this on a private range with all this stuff that is written on this www.process-one.net website.
